common/fixation_entry.go

```diff
@@ -60,15 +61,60 @@ func GetEntry(ctx sdk.Context, storeKey sdk.StoreKey, entryKeyPrefix string, cdc
 	}
 
 	cdc.MustUnmarshal(b, &val)
+
+	switch referenceAction {
+	case types.ADD_REFERENCE:
+		val.References += 1
+		SetEntry(ctx, storeKey, entryKeyPrefix, cdc, val)
+	case types.SUB_REFERENCE:
+		if val.GetReferences() > 0 {
+			val.References -= 1
+			SetEntry(ctx, storeKey, entryKeyPrefix, cdc, val)
+		}
+	case types.DO_NOTHING:
+	}
+
 	return val, true
 }
 
+// Helper function to extract the original index from an older version index (for example: myIndex_2 -> myIndex)
+func extractOriginalIndexFromOlderVersionIndex(index string) (originalIndexWithoutVersionNumSuffix string) {
```

Comment:
this can cause problems, we dont want to play with strings like that

Reply:
done